packages/js-dapi-client/lib/BlockHeadersProvider/BlockHeadersProvider.js

```diff
@@ -55,67 +69,184 @@ class BlockHeadersProvider extends EventEmitter {
 
   /**
    *
-   * @param spvChain
+   * @param {SpvChain} spvChain
    */
   setSpvChain(spvChain) {
     this.spvChain = spvChain;
   }
 
-  async start() {
-    if (!this.coreMethods) {
-      throw new Error('Core methods have not been provided. Please use "setCoreMethods"');
-    }
-
-    if (this.started) {
-      throw new Error('BlockHeaderProvider has already been started');
-    }
+  /**
+   * @private
+   */
+  initReader() {
+    if (!this.blockHeadersReader) {
```

Comment:
Move to to createBlockHeadersProvider fabric

Reply:
Done